model/user.go

```diff
@@ -47,6 +47,77 @@ const (
 	USER_PASSWORD_MAX_LENGTH  = 72
 )
 
+// This is a struct for a user interaction
+// A user interaction will represent one row in the results of the
+// GetUserInteractions query in store/sqlstore/user_store.go
+type UserInteraction struct {
+	// The type of interaction
+	// enum('Reaction', 'Mention', 'Reply', 'DirectMessage')
+	InteractionType string `json:"interaction_type"`
+
+	// The username of the user that this interaction took place with
+	Username string `json:"username"`
+
+	// Denotes whether or not the current user was the recipient of this interaction
+	IsRecipient bool `json:"is_recipient"`
+
+	// The context in which this interaction occurred (either 'Course' or the learning group type's prefix (ex. 'plg'))
+	Context string `json:"context"`
+
+	// The type of channel that this interaction took place in
+	// enum('P', 'O', 'D', 'G')
+	//     P = private channel
+	//     O = public channel
+	//     D = direct message (1 on 1)
+	//     G = group direct message
+	ChannelType string `json:"channel_type"`
+
+	// The slug name of the channel that this interaction took place in (Channels.Name)
+	ChannelName string `json:"channel_name"`
+}
+
+// This is a struct for a learning group type
+// Each learning group type is derived from personal channel names found in the config
+type LearningGroupType struct {
+	// ex. 'capstone'
+	Prefix string `json:"prefix"`
+}
+
+// This is a struct for all user analytics data sets
+// This is the struct that gets returned to mattermost-redux in the users/{userId}/teams/{teamId}/useranalytics request
```

Comment:
Another note, mattermost-redux is a client of the mattermost-server api but it is not the only client. The api calls can be used w/ anyone w/ the proper url (and hopefully authorization), so mentions mattermost-redux as the specific recipient is probably wrong.
That said, I don't care enough to insist it be changed, but it was something I felt should be said so you think about it in the future.